test/e2e/quadlet_test.go

```diff
@@ -660,7 +660,12 @@ var _ = Describe("quadlet system generator", func() {
 			}
 
 			// Run quadlet to convert the file
-			session := podmanTest.Quadlet([]string{"--user", "--no-kmsg-log", generatedDir}, quadletDir)
+			var args []string
+			if isRootless() {
+				args = append(args, "--user")
+			}
+			args = append(args, "--no-kmsg-log", generatedDir)
```

Comment:
style question: why not `args := []string{"--no-kmsg-log", generatedDir}` (initialize first) followed by prepending `--user` if rootless?

Reply:
because `generatedDir` is a argument and quadlet does not seem to accept options/flag after the args
```
$ bin/quadlet --dryrun --no-kmsg-log --user /tmp 
quadlet-generator[162564]: Loading source unit file /home/pholzing/.config/containers/systemd/t1.container
...
$ bin/quadlet --dryrun --no-kmsg-log /tmp --user
quadlet-generator[162630]: No files parsed from [/run/containers/systemd /etc/containers/systemd /usr/share/containers/systemd]
```
The --user is ignored but I haven't looked into why, the systemd generator has a forced syntax defined by systemd so I just accepted it without thinking to long about it.

Reply:
FWIW `podman` also has assumption about the order between options and arguments - all options must come first, then the first argument being the image and everything after it goes into the command. So, maybe it's a `cobra` thing.
Having said that, as @edsantiago wrote, the `--user` parameter can be prepended. Not that critical I guess

Reply:
> FWIW podman also has assumptions about the order between options and arguments - all options must come first, then the first argument being the image and everything after it goes into the command. So, maybe it's a cobra thing.

quadlet does not use cobra, it used the std `flag` package. In cobra you control that via `flags.SetInterspersed(true/false)` and yes a command like `podman run $IMAGE` treats all options after $IMAGE as args to the container command but in other commands it is fine to mix them.

Reply:
Oh, I didn't remember that. I guess the different package is the reason.
Also, AFAIK systemd does not pass any options only parameters. Quadlet uses `argv[0]` to set `isUser` (though it should do that using an environment variable). So, the options are only for debugging and testing